components/net/fetch/request.rs

```diff
@@ -574,15 +573,19 @@ fn http_network_or_cache_fetch(request: Rc<RefCell<Request>>,
 
     // Step 5
     if let Some(content_length_value) = content_length_value {
-        http_request.borrow_mut().headers.set(ContentLength(content_length_value));
+        http_request.headers.borrow_mut().set(ContentLength(content_length_value));
     }
 
     // Step 6
-    match http_request.borrow().referer {
+    let referer;
+    {
+        referer = http_request.referer.clone();
+    }
```

Comment:
That's weird, I swear I ran into this and fixed it. Must've been a similar legacy handling from elsewhere. I've updated it now- no clone is needed at all! Cloning was just to get around pointer lifetimes.